lib/ood_core/job/adapters/linux_systemd/launcher.rb

```diff
@@ -163,10 +163,15 @@ def export_env(script)
 
   def script_timeout(script)
     wall_time = script.wall_time.to_i
-    return site_timeout if wall_time == 0
-    return [wall_time, site_timeout].min unless site_timeout == 0
-
-    wall_time
+    if wall_time == 0
+      # this is the only way it can be 0
+      # so make it into infinify for systemd to never terminate
+      site_timeout == 0 ? 'infinity' : site_timeout
```

Comment:
ohhhh. @treydock do we want to allow infinity?  I'm kinda thinking yes even though sites may not expect it.

At least as unit files admins can find and terminate them easily, so they're discoverable (and not just through ps + grep).

Reply:
Well actually we already allow it in the `linux_host` adapter (though I don't know if we knew that at the time).

It's just once we enable it we can't disable it without some pain for folks, so I want to be sure we're purposeful about it.

Reply:
I liked you pointing it out because I think some sites will want that - run until machine gets rebooted. Overall it is unlikely to be a problem because the default for min on bc_num_hours attribute is 1 - you have to go out of your way to allow 0. That is - as long as it is documented.

As far as how discoverable it is - you'd be surprised. I find it pretty hard to discover user units as root (by using systemctl).